NeosModLoader/Utility/GenericTypeMethodsInvoker.cs

```diff
@@ -0,0 +1,295 @@
+using HarmonyLib;
+using NeosModLoader;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NeosModLoader.Utility
+{
+	/// <summary>
+	/// Represents a generic method invoker that invokes potentially generic methods on generic types.
+	/// </summary>
+	public sealed class GenericTypeMethodsInvoker
```

Comment:
Could a short [`<example><code>`](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments#d33-code), since this seems to be the "entry point" of this utility API? 😜 But yeah at least some docs is better than no docs 👍